state/execution.go

```diff
@@ -254,6 +254,30 @@ func (blockExec *BlockExecutor) Commit(
 	return res.Data, res.RetainHeight, err
 }
 
+func (blockExec *BlockExecutor) GetCheckBlockResp(block *types.Block) (*abcix.ResponseCheckBlock, error) {
```

Comment:
from api design’s point of view, we should probably simplify the return type here.